src/System.Private.CoreLib/shared/System/Globalization/CultureInfo.cs

```diff
@@ -1287,24 +1175,53 @@ public static CultureInfo GetCultureInfo(string name)
         /// </summary>
         public static CultureInfo GetCultureInfo(string name, string altName)
         {
-            if (name == null)
+            if (name is null)
             {
                 throw new ArgumentNullException(nameof(name));
             }
-            if (altName == null)
+            if (altName is null)
             {
                 throw new ArgumentNullException(nameof(altName));
             }
 
-            CultureInfo? retval = GetCultureInfoHelper(-1, name, altName);
-            if (retval == null)
+            name = CultureData.AnsiToLower(name);
+            altName = CultureData.AnsiToLower(altName);
+            string nameAndAltName = name + "\xfffd" + altName;
+            Dictionary<string, CultureInfo> nameTable = CachedCulturesByName;
+            CultureInfo? result;
+
+            lock (nameTable)
+            {
+                if (nameTable.TryGetValue(nameAndAltName, out result))
+                {
+                    return result;
+                }
+            }
+
+            try
+            {
+                result = new CultureInfo(name, altName) { _isReadOnly = true };
+                result.TextInfo.SetReadOnlyState(readOnly: true); // TextInfo object is already created; we need to set it as read only.
+            }
+            catch (ArgumentException)
+            {
+                throw new CultureNotFoundException("name or altName", SR.Format(SR.Argument_OneOfCulturesNotSupported, name, altName));
```

Comment:
Maybe pedantic, but if it has the English word 'or' maybe it should be in the resx.

Reply:
`name/altName` ?

Reply:
Yeah.  Pre-existing, though, and I wasn't sure of the history here.

Reply:
I think the first parameter is just the param name and not a real exception message. as @stephentoub split the method, we can just use the correct param name.

Reply:
> we can just use the correct param name

Which one is correct?  We're passing both name and altName to the CultureInfo ctor.

Reply:
name/altName would be ok I guess.